crates/pathfinder/src/storage.rs

```diff
@@ -145,6 +146,170 @@ fn migrate_from_0_to_1(transaction: &Transaction) -> anyhow::Result<()> {
         .context("Failed to migrate StarkNet state tables to version 1")
 }
 
+fn migrate_to_2(tx: &Transaction) -> anyhow::Result<()> {
+    use sha3::{Digest, Keccak256};
+
+    // we had a mishap of forking the schema at version 1 so to really support all combinations of
+    // schema at version 1 we need to make sure that contracts table still looks like:
+    // CREATE TABLE contracts (
+    //     address    BLOB PRIMARY KEY,
+    //     hash       BLOB NOT NULL,
+    //     bytecode   BLOB,
+    //     abi        BLOB,
+    //     definition BLOB
+    // );
+
+    {
```

Comment:
Cool that you're now distinguishing between the "old" v1 and "new" v1, :sweat_smile: